Applications/Spire/Source/Ui/Window.cpp

```diff
@@ -163,14 +180,10 @@ bool Window::nativeEvent(const QByteArray& eventType, void* message,
     return true;
   } else if(msg->message == WM_SIZE) {
     if(msg->wParam == SIZE_MAXIMIZED) {
-      auto abs_pos = mapToGlobal(m_title_bar->pos());
-      auto pos = QGuiApplication::screenAt(mapToGlobal({window()->width() / 2,
-        window()->height() / 2}))->geometry().topLeft();
-      pos = QPoint(std::abs(abs_pos.x() - pos.x()),
-        std::abs(abs_pos.y() - pos.y()));
-      setContentsMargins(pos.x(), pos.y(), pos.x(), pos.y());
+      layout()->setContentsMargins({});
```

Comment:
Why do we need to change margins when maximizing or restorting the window?

Reply:
We simulate a border by setting the Window's background color and then setting the content layout's margins to make a small part of the background visible. When the Window is maximized we remove these margins to hide the border, then restore them when the Window is restored so that the border is visible again.

If we added a stylesheet for the Window border we would still need margins to make it visible (because of the Window body), so it seems easier to just define the border using the Window background color.